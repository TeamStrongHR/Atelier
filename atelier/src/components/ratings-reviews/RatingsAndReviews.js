import ReviewsList from './components/ReviewsList.js';
import Ratings from './components/Ratings.js';
import axios from 'axios';
import {useState, useEffect} from 'react';


export default function RatingsAndReviews () {

  useEffect(()=> {
    var options = {
      url: "http://localhost:3000/api/reviews",
      params: {
        page: 1,
        count: 6,
        sort: "newest",
        product_id: 37311
      },
      method: "get",
    }
    axios(options)
    .then(result => {
      console.log(result);
    })
  })

  return (
    <div data-testid="ratings-reviews-comp" className="ratings-reviews">
      <Ratings/>
      <ReviewsList />
    </div>
  )
}