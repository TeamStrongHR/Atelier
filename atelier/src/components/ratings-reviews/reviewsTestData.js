export var reviews = {
  "product": "37313",
  "page": 0,
  "count": 5,
  "results": [
      {
          "review_id": 1277202,
          "rating": 3,
          "summary": "Test Submit",
          "recommend": false,
          "response": null,
          "body": "Apsafdjkabflkjhbsdkjafhbdkjlhbflkdsabkvldbajkvavdfsdvsdvds",
          "date": "2022-10-25T00:00:00.000Z",
          "reviewer_name": "Rendy",
          "helpfulness": 3,
          "photos": [
              {
                  "id": 2456480,
                  "url": "https://res.cloudinary.com/dvijvlkad/image/upload/ja6cv28co6xuyb1nvk9k"
              }
          ]
      },
      {
          "review_id": 1277203,
          "rating": 3,
          "summary": "Test Submit",
          "recommend": false,
          "response": null,
          "body": "Apsafdjkabflkjhbsdkjafhbdkjlhbflkdsabkvldbajkvavdfsdvsdvds",
          "date": "2022-10-25T00:00:00.000Z",
          "reviewer_name": "Rendy",
          "helpfulness": 2,
          "photos": [
              {
                  "id": 2456481,
                  "url": "https://res.cloudinary.com/dvijvlkad/image/upload/ja6cv28co6xuyb1nvk9k"
              }
          ]
      },
      {
          "review_id": 1277205,
          "rating": 3,
          "summary": "HAPPY HAPPY HAPPY",
          "recommend": false,
          "response": null,
          "body": "THIS PRODUCT IS AWESOME WOOHOOO THIS IS AWESOME",
          "date": "2022-10-25T00:00:00.000Z",
          "reviewer_name": "Ray",
          "helpfulness": 2,
          "photos": []
      },
      {
          "review_id": 1277207,
          "rating": 3,
          "summary": "gfdsfdsgasdgdsagsdgsa",
          "recommend": true,
          "response": null,
          "body": "fdasasfasgadgdsagdgdfsaaaaaaaaaaaaaaaaaaaaa",
          "date": "2022-10-25T00:00:00.000Z",
          "reviewer_name": "gsagdsag",
          "helpfulness": 1,
          "photos": []
      },
      {
          "review_id": 1277206,
          "rating": 4,
          "summary": "Whats up hahahaha",
          "recommend": false,
          "response": null,
          "body": "This product is super duper cool and i lkove it ",
          "date": "2022-10-25T00:00:00.000Z",
          "reviewer_name": "Rendy",
          "helpfulness": 1,
          "photos": []
      }
  ]
}

export var breakdown = {
  "product_id": "37313",
  "ratings": {
      "1": "5",
      "2": "3",
      "3": "18",
      "4": "24",
      "5": "25"
  },
  "recommended": {
      "false": "16",
      "true": "59"
  },
  "characteristics": {
      "Fit": {
          "id": 125036,
          "value": "2.6666666666666667"
      },
      "Length": {
          "id": 125037,
          "value": "2.7903225806451613"
      },
      "Comfort": {
          "id": 125038,
          "value": "2.8730158730158730"
      },
      "Quality": {
          "id": 125039,
          "value": "3.1269841269841270"
      }
  }
}
