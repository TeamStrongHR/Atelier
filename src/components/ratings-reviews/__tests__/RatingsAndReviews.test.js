import {render, screen, cleanup} from '@testing-library/react';
import renderer from 'react-test-renderer';
import RatingsAndReviews from '../RatingsAndReviews.js';
import {reviews, breakdown} from '../reviewsTestData.js';
import axios from 'axios';


afterEach(() => {
  cleanup()
});

jest.mock('axios', () => ({
  __esModule: true,
  get: jest.fn(() => Promise.resolve({ data: "data" })),
  default: jest.fn(() => Promise.resolve({ data: 'data' })),
}));

describe("RatingsAndReviews Component", ()=> {
  render(<RatingsAndReviews product_id={37311} productName={'Jeans'} />);
    const container = screen.getByTestId('ratings-reviews-comp')

  test("Should render RatingsAndReviews component", ()=> {
    expect(container).toBeInTheDocument();
  })

  test("It should render ReviewList component", ()=> {
    expect(container.getElementsByClassName("reviews-list")).toBeDefined();
  })

  test('matches snapshot', ()=> {
    const tree = renderer.create(<RatingsAndReviews product_id={37311} productName={'Jeans'} />).toJSON();
    expect(tree).toMatchSnapshot();
  })
})

