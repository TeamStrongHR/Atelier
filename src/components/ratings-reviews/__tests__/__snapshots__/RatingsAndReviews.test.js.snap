// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RatingsAndReviews Component matches snapshot 1`] = `
<div
  className="ratings-reviews"
  data-testid="ratings-reviews-comp"
/>
`;
